packages/ckeditor5-table/src/tableutils.js

```diff
@@ -753,17 +757,19 @@ function adjustHeadingColumns( table, removedColumnIndexes, writer ) {
 
 // Calculates a new heading rows value for removing rows from heading section.
 function updateHeadingRows( table, first, last, model, batch ) {
-	const headingRows = table.getAttribute( 'headingRows' ) || 0;
+	// Must be done after the changes in table structure (removing rows).
+	// Otherwise the downcast converter for headingRows attribute will fail. ckeditor/ckeditor5#6391.
+	//
```

Comment:
```suggestion
	// Otherwise the downcast converter for headingRows attribute will fail.
	// See https://github.com/ckeditor/ckeditor5/issues/6391.
	//
```